src/fuzzfetch/fetch.py

```diff
@@ -457,12 +457,12 @@ def __init__(self, target, branch, build, flags, platform=None, nearest=None):
                 # If start date is outside the range of the newest/oldest available build, adjust it
                 if asc:
                     start = max(start, now - timedelta(days=364))
-                    end = now
+                    end = min(start, now)
```

Comment:
Is this not normally going to be `end = start` ? `start` will usually be less than `now`, so you'll end up not searching at all.

Reply:
You're right.  End should always be now but I think we still need to check to ensure that start isn't in the future.  I'll fix this in a separate commit.

Reply:
So maybe `start = min(max(start, now - timedelta(days=364)), now)` and `end = now`